File: lyric/src/main/java/com/android/lyric/ILrcBuilder.java
package com.android.lyric;

import com.android.lyric.impl.LrcRow;

import java.util.List;

/**
 * 解析歌词，得到LrcRow的集合
 */
public interface ILrcBuilder {
    /**
     * 解析歌词，得到LrcRow的集合
     *
     * @param rawLrc lrc内容
     * @return LrcRow的集合
     */
    List<LrcRow> getLrcRows(String rawLrc);
}

File: lrc/src/main/java/com/android/wy/news/lrc/listener/ILrcBuilder.kt
package com.android.wy.news.lrc.listener

import com.android.wy.news.lrc.impl.LrcRow

/*     
  * @Author:         gao_yun
  * @CreateDate:     2023/5/22 9:52
  * @Version:        1.0
  * @Description:    
 */
interface ILrcBuilder {
    /**
     * 解析歌词，得到LrcRow的集合
     *
     * @param rawLrc lrc内容
     * @return LrcRow的集合
     */
    fun getLrcRows(rawLrc: String?): ArrayList<LrcRow>?
}